tiledb/sm/subarray/subarray.cc

```diff
@@ -1011,7 +1011,20 @@ Status Subarray::compute_tile_overlap() {
         } else {  // Sparse fragment
           auto rtree = (const RTree*)nullptr;
           RETURN_NOT_OK(meta[i]->rtree(*encryption_key, &rtree));
-          tile_overlap_[i][j] = rtree->get_tile_overlap<T>(range);
+          if (rtree != nullptr) {  // Format version > 2
+            tile_overlap_[i][j] = rtree->get_tile_overlap<T>(range);
+          } else {  // Format version <= 2 - use linear scan over MBRs
+            auto mbrs = meta[i]->mbrs();
```

Comment:
probably would be good to consolidate into an external function

Reply:
Yeah, I can put this into a different function as well.